### PR TITLE
Make crosshair dot centered

### PR DESCRIPTION
**Pull request by a7f3**
## before
image
## after
image

### CODE REVIEW

File: gamemodes/terrortown/entities/weapons/weapon_tttbase.lua [hunk 1]
```diff
@@ -360,8 +360,9 @@ if CLIENT then
 
 		-- draw crosshair dot
 		if cvCrosshairMode:GetInt() == CROSSHAIR_MODE_DOT_AND_LINES or cvCrosshairMode:GetInt() == CROSSHAIR_MODE_DOT_ONLY then
-			local xDot = mathFloor(xCenter - thicknessLine * 0.5)
-			local yDot = mathFloor(yCenter - thicknessLine * 0.5)
+			local xDot = xCenter - mathFloor(thicknessLine * 0.5)
+			local yDot = yCenter - mathFloor(thicknessLine * 0.5)
```
Comment (ZenBre4ker) [inline gamemodes/terrortown/entities/weapons/weapon_tttbase.lua:364]: Doesnt that mean a thicknessLine of 1 is invisible?

Comment (a7f3) [reply gamemodes/terrortown/entities/weapons/weapon_tttbase.lua:364]: this is just for the dot's position, if the thicknessLine is 1, it means there's no offset

**Comment by EntranceJew** [comment]
i need to double check this when i have a moment later but iirc i think at certain scale combinations and outline weights it is still possible to offset the line weights to a similar effect, i think this PR could be an opportunity to fix that also

**Comment by EntranceJew** [comment]
image

image

it's a little hard to see but with the right settings i think the lines are misaligned also, but not nearly as bad as they were before without integer multipliers

**Comment by ZenBre4ker** [comment]
> it's a little hard to see but with the right settings i think the lines are misaligned also, but not nearly as bad as they were before without integer multipliers

I think you are right, in the code you can see that in x and y direction the `thicknessOutline` is subtracted for all 4 rectangles. This does not make sense for one direction per rectangle.
For example the left and right rectangle, dont need the `thicknessOutline` subtracted for the y direction (height), as it should have no effect on the center.
Same goes for upper and lower rect in x direction (width).

**Review by ZenBre4ker — APPROVED** [review]
This change is good and is also theoretically more precise, as we also only floor for the offsetLine, which is subtracted from the center calculation for the lines.

File: gamemodes/terrortown/entities/weapons/weapon_tttbase.lua [hunk 2]
```diff
@@ -360,8 +360,8 @@ if CLIENT then
...
```
Comment (ZenBre4ker) [inline gamemodes/terrortown/entities/weapons/weapon_tttbase.lua:364]: Just use the offsetLine variable that has these values already calculated.

**Review by ZenBre4ker — CHANGES_REQUESTED** [review]
Actually just use the variable.

**Review by TimGoll — APPROVED** [review]
It is fine once @ZenBre4ker's comment is addressed. Thank you!

File: gamemodes/terrortown/entities/weapons/weapon_tttbase.lua (hunk 2 above)
Comment (EntranceJew) [reply gamemodes/terrortown/entities/weapons/weapon_tttbase.lua:364]:
```suggestion
			local xDot = xCenter - offsetLine
			local yDot = yCenter - offsetLine
```
should just be able to click this to make that change afaik

**Review by ZenBre4ker — APPROVED** [review]

**Pull request merged**